cmd/up.go

```diff
@@ -158,41 +183,141 @@ func (cmd *UpCmd) Run(ctx context.Context, devPodConfig *config.Config, client c
 		log.Infof("Run 'ssh %s.devpod' to ssh into the devcontainer", client.Workspace())
 	}
 
+	dotfilesRepo := devPodConfig.ContextOption(config.ContextOptionDotfilesURL)
+	if cmd.DotfilesSource != "" {
+		dotfilesRepo = cmd.DotfilesSource
+	}
+
+	dotfilesScript := devPodConfig.ContextOption(config.ContextOptionDotfilesScript)
+	if cmd.DotfilesScript != "" {
+		dotfilesScript = cmd.DotfilesScript
+	}
+
+	if dotfilesRepo != "" {
+		log.Infof("Dotfiles repo %s specified", dotfilesRepo)
+		log.Infof("Setting dotfiles into the devcontainer")
+
+		execPath, err := os.Executable()
+		if err != nil {
+			return err
+		}
+
+		agentArguments := []string{
+			"--debug",
+			"agent",
+			"workspace",
+			"install-dotfiles",
+			"--repository",
+			dotfilesRepo,
+		}
+
+		if dotfilesScript != "" {
+			log.Infof("Dotfiles script %s specified", dotfilesScript)
+
+			agentArguments = append(agentArguments, "--install-script")
+			agentArguments = append(agentArguments, dotfilesScript)
+		}
+
+		dotCmd := exec.Command(
+			execPath,
+			[]string{
+				"ssh",
+				"--debug",
+				"--agent-forwarding=true",
+				"--start-services=false",
+				"--context",
+				client.Context(),
+				client.Workspace(),
+				"--log-output=raw",
+				"--command",
+				agent.ContainerDevPodHelperLocation + " " +
+					strings.Join(agentArguments, " "),
+			}...,
+		)
+
+		log.Debugf("Running command: %v", dotCmd.Args)
+
+		writer := log.Writer(logrus.DebugLevel, false)
```

Comment:
Please make this info level, would be good to see the dotfiles script output by default

Reply:
Done :+1: